olive/passes/onnx/gptq.py

```diff
@@ -0,0 +1,165 @@
+                type_=bool,
+                default_value=False,
+                description="Use descriptive activation for quantization. Default value is False.",
+            ),
+            "sym": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Symmetric quantization. Default value is False.",
+            ),
+            "gpu": PassConfigParam(
+                type_=bool,
+                default_value=True,
+                description="Use GPU for quantization. Default value is True.",
+            ),
+            "export_optimization": PassConfigParam(
+                type_=str,
+                default_value=None,
+                description="Export optimization level. Default value is None.",
+            ),
+        }
+
+    @torch.no_grad()
+    def _run_for_config(
+        self, model: PyTorchModelHandler, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> ONNXModelHandler:
+        from onnxruntime import __version__ as ort_version
+        from optimum.exporters.onnx import onnx_export_from_model
+        from optimum.version import __version__ as optimum_version
+        from transformers import AutoModelForCausalLM, AutoTokenizer, GPTQConfig
+
+        from olive.passes.onnx.gptq_utils import QuantLinearORT
+
+        if version.parse(optimum_version) < version.parse("1.17.0"):
+            raise ValueError("Please use optimum>=1.17.0 for gptq quantization")
+        if version.parse(ort_version) < version.parse("1.17.0"):
+            raise ValueError("Please use onnxruntime-gpu>=1.17.0 for gptq quantization")
+
+        tokenizer = AutoTokenizer.from_pretrained(model.hf_config.model_name, use_fast=False)
+
+        gptq_config = GPTQConfig(
+            bits=config["bits"],
+            dataset=config["dataset"],
+            group_size=config["group_size"],
+            damp_percent=config["damp_percent"],
+            static_groups=config["static_groups"],
+            true_sequential=config["true_sequential"],
+            desc_act=config["desc_act"],
+            sym=config["sym"],
+            tokenizer=tokenizer,
+        )
+
+        def get_onnx_quant_linear(*args, **kwargs):
+            return QuantLinearORT
+
+        import auto_gptq
+
+        original = auto_gptq.utils.import_utils.dynamically_import_QuantLinear
+        try:
+            # GPTQ Quantization in transformers use auto_gptq under the hood, so we
+            # replace QuantLinear in auto_gptq with QuantLinearORT for quant linear layer packing
+            auto_gptq.utils.import_utils.dynamically_import_QuantLinear = get_onnx_quant_linear
+            quantized_model = AutoModelForCausalLM.from_pretrained(
+                model.hf_config.model_name,
+                quantization_config=gptq_config,
+                attn_implementation="eager",
+                torch_dtype=torch.half,
+            )
+        except ValueError:
```

Comment:
if it is only for logging purpose and re-raise, then the except is needless.

Reply:
Yeah, I think transformers will give accurate error info about the model_name_or_path parameter, the exception here to catch this error may not help much. Is that ok for your [previous suggestion](https://github.com/microsoft/Olive/pull/957#discussion_r1503563880)? @xiaoyu-work

Reply:
I'd like to add some Olive specific error messages like `"{model.hf_config.model_name} is not supported. GptqQuantizer pass only supports auto-regressive language models."`

Reply:
for this kinds of ValueError or general exception, there is no way to figure out whether it is parameter issue or model is not supported since it is too general exception type.

On the other hand, only add logging and rethrown exception is needless since the caller will catch it and handle it.